Multisig/Logic/Ethereum/BlockchainDomainManager.swift

```diff
@@ -77,10 +78,95 @@ class BlockchainDomainManager {
             return GSError.UDUnregisteredName()
         case .unspecifiedResolver:
             return GSError.UDResolverNotFound()
+        case .unknownError(let internalError):
+
+            switch internalError {
+            case APIError.decodingError:
+                return GSError.UDDecodingError()
+            case APIError.encodingError:
+                return GSError.UDEncodingError()
+            case let detailedError as DetailedLocalizedError:
+                return detailedError
+            default:
+                break
+            }
+
+            fallthrough
         default:
             return GSError.ThirdPartyError(
                 reason: error.localizedDescription
             )
         }
     }
 }
+
+// Taken from the DefaultNetworkingLayer implementation to replace
+// the error handling with more descriptive error from HTTTPClient
+public struct GSNetworkingLayer: NetworkingLayer {
```

Comment:
would it be better to create a PR into their repo to return better errors?

Reply:
I wouldn't want to spend more time on it now, it's good enough